apps/ledger-live-mobile/src/screens/FirmwareUpdate/index.tsx

```diff
@@ -0,0 +1,446 @@
+        break;
+      default:
+        break;
+    }
+
+    return Object.values(newSteps);
+  }, [
+    defaultSteps.installUpdate,
+    defaultSteps.prepareUpdate,
+    defaultSteps.restoreAppsAndSettings,
+    t,
+    totalNumberOfSteps,
+    updateState.progress,
+    updateState.step,
+  ]);
+
+  const deviceInteractionDisplay = useMemo(() => {
+    if (updateState.error !== null) {
+      return (
+        <DeviceActionError
+          device={device}
+          t={t}
+          errorName={updateState.error.name}
+          translationContext="FirmwareUpdate"
+        >
+          <Button
+            type="main"
+            outline={false}
+            onPress={quitUpdate}
+            mt={6}
+            alignSelf="stretch"
+          >
+            {t("FirmwareUpdate.quitUpdate")}
+          </Button>
+        </DeviceActionError>
+      );
+    }
+
+    switch (updateState.step) {
+      case "allowSecureChannelRequested":
+        return (
+          <AllowManager
+            device={device}
+            wording={t("DeviceAction.allowSecureConnection")}
+          />
+        );
+      case "installOsuDevicePermissionRequested":
+        return (
+          <ConfirmFirmwareUpdate
+            device={device}
+            oldFirmwareVersion={deviceInfo.seVersion ?? ""}
+            newFirmwareVersion={firmwareUpdateContext.final.name}
+            t={t}
+          />
+        );
+      case "installOsuDevicePermissionDenied":
+        return (
+          <FirmwareUpdateDenied
+            device={device}
+            newFirmwareVersion={firmwareUpdateContext.final.name}
+            onPressRestart={triggerUpdate}
+            onPressQuit={quitUpdate}
+            t={t}
+          />
+        );
+      case "installOsuDevicePermissionGranted":
+        if (!firmwareUpdateContext.shouldFlashMCU) {
+          return <FinishFirmwareUpdate device={device} t={t} />;
+        }
+        return undefined;
+      case "flashingMcu":
+        if (updateState.progress === 1) {
+          return <FinishFirmwareUpdate device={device} t={t} />;
+        }
+        return undefined;
+      default:
+        return undefined;
+    }
+  }, [
+    updateState.error,
+    updateState.step,
+    updateState.progress,
+    device,
+    deviceInfo.seVersion,
+    firmwareUpdateContext.final.name,
+    firmwareUpdateContext.shouldFlashMCU,
+    t,
+    triggerUpdate,
+    quitUpdate,
+  ]);
+
+  return (
+    <>
+      <Flex flex={1} justifyContent="space-between">
+        <Flex>
+          <Text variant="h4" ml={5}>
+            {t("FirmwareUpdate.updateDevice", { deviceName })}
+          </Text>
+          <VerticalStepper steps={steps} />
+        </Flex>
+        <Flex mx={6} mb={8}>
+          <Alert type="info" title={t("FirmwareUpdate.doNotLeaveLedgerLive")} />
+        </Flex>
+      </Flex>
+      <QueuedDrawer
+        isRequestingToBeOpened={Boolean(deviceInteractionDisplay)}
+        noCloseButton
+      >
+        {deviceInteractionDisplay}
```

Comment:
Only changing the content of the `QueuedDrawer` does not: add a new drawer to the queue, close the current drawer, display the new drawer - it just changes the content of the current displayed drawer

Is it rendering correctly/as you expect ?

As long as you don't enable a close button and do something on the `onClose` of the QueuedDrawer that would do different thing depending on the `updateState.step` you're good

Reply:
It's ok, there shouldn't be a queue here. We can review that if we end up needing a close button.